python/ray/util/data/interface.py

```diff
@@ -0,0 +1,26 @@
+from typing import Iterable
+
+import pandas as pd
+
+
+class SourceShard:
+    @property
+    def shard_id(self) -> int:
+        raise NotImplementedError
+
+    def __iter__(self) -> Iterable[pd.DataFrame]:
+        raise NotImplementedError
+
+    def __str__(self):
+        return repr(self)
+
+    def __repr__(self):
+        return f"SourceShard[{self.shard_id}]"
+
+
+class TensorDataset:
```

Comment:
Do we need this interface?

Reply:
TensorDataset is a abstract common interface for `TorchMLDataset` and `TFMLDataset`, this is as @richardliaw comments.

Reply:
Hmm, it seems there isn't a lot of added value, why not have the two datasets directly implement MLDataset to avoid the extra abstraction?

Reply:
Will refactor it

Reply:
Hi @ericl, MLDataset is inherited from `ParallelIterator` and limit the record as `pandas.DataFrame`.  If the `TorchMLDataset` and `TFMLdataset` inherit from `MLDataset`, we still need to rewrite those methods due to the different record type.  Is it as same as inherited from `ParallelIterator`?

```
                       ParallelIterator
                   
                    /          |              \
      TorchMLDataset  <---  MLDataset  ----> TFMLDataset
```

The above is excepted?

Reply:
OK so it seems like `MLDataset` has this explicit inheritance:

`class MLDataset(ParallelIterator[pd.DataFrame])` which is explicit about what it provides/handles.

Thus, I think it doesn't make it a base class for Torch/TF. 

Should we just remove this TensorDataset interface then and merge @ericl?

Reply:
Sounds good. So all three classes inherit directly from ParallelIterator

Reply:
(or, as discussed offline, remove inheritance for Torch/TF)

Reply:
thanks all, removed the `TensorDataset` interface.